extends CanvasLayer
@onready var texture_rect = $Texture

const SCENE_MAIN_MENU = "res://scenes/main_menu/main_menu.tscn"
const SCENE_LEVEL = "res://scenes/level/level.tscn"

@onready var animation_player = $AnimationPlayer

func _ready():
	self.texture_rect.hide()

func change_scene(target: String) -> void:
	self.get_tree().paused = true
	#animation_player.play("fade_out")
	#await animation_player.animation_finished
	await self.set_screenshot()
	self.fade_in()
	self.fade_out()
	self.get_tree().change_scene_to_file(target)
	self.get_tree().paused = false

func fade_in():
	self.texture_rect.show()
	self.texture_rect.modulate = Color.WHITE

func fade_out():
	var tween = create_tween().set_trans(Tween.TRANS_SINE).set_ease(Tween.EASE_IN)
	tween.tween_property(self.texture_rect, "modulate", Color.TRANSPARENT, 0.25)
	tween.tween_callback(self.texture_rect.hide)

func set_screenshot():
	var capture = get_viewport().get_texture().get_image()
	texture_rect.texture = ImageTexture.create_from_image(capture)
